### PR TITLE
refactor: remove remaining vararg log messages

### PR DESCRIPTION
**Pull request by ckerr**
Part 8 of a series to update Transmission's logging mechanism. #2774 is the previous in the series. The goals are described [here](https://github.com/transmission/transmission/pull/2749#pullrequest-875916992).

---

This PR finishes off the task of replacing all the varargs log messages with [fmt](https://fmt.dev) for type safety + simplicity.

Another refactor is to remove the `LogName` / `CodeName` idiom that was used in previous PRs in this series. That value was always a variation on the filename that the log message came from, and that filename was already being passed in via the `file` argument.  So `tr_logAddFoo()` now takes a message `std::string_view` and an optional name `std::string_view`. If the name is empty, a default one is built from `basename(file)`.

### CODE REVIEW

**Comment by ckerr** [comment]
@mikedld review welcome, but FWIW this is mostly code cleanup.

There are some translatable string changes, but not the magnitude of the previous PRs. :smile_cat:

File: libtransmission/log.h [hunk 1]
```diff
@@ -5,47 +5,102 @@
...
+[[nodiscard]] bool tr_logLevelIsActive(tr_log_level level);
+
+////
+
+void tr_logAddMessage(
+    char const* source_file,
+    int source_line,
+    tr_log_level level,
+    std::string_view message,
+    std::string_view module_name = "");
```
Comment (mikedld) [inline libtransmission/log.h:86]:
```suggestion
    std::string_view module_name = {});
```

File: libtransmission/port-forwarding.cc [hunk 2]
```diff
@@ -93,12 +91,10 @@ static void natPulse(tr_shared* s, bool do_check)
...
         session->public_peer_port = public_peer_port;
         session->private_peer_port = received_private_port;
-        tr_logAddNamedInfo(
-            LogName,
-            fmt::format(
-                _("Mapped private port '{private_port}' to public port '{public_port}'"),
-                fmt::arg("public_port", session->public_peer_port),
-                fmt::arg("private_port", session->private_peer_port)));
+        tr_logAddInfo(fmt::format(
+            _("Mapped private port '{private_port}' to public port '{public_port}'"),
```
Comment (mikedld) [inline libtransmission/port-forwarding.cc:95]:
```suggestion
            _("Mapped private port {private_port} to public port {public_port}"),
```

File: libtransmission/torrent-magnet.cc [hunk 3]
```diff
@@ -313,7 +311,7 @@ static void onHaveAllMetainfo(tr_torrent* tor, tr_incomplete_metadata* m)
 
         m->piecesNeededCount = n;
         char const* const msg = error != nullptr && error->message != nullptr ? error->message : "unknown error";
-        logdbg(tor, fmt::format("metadata error: {}. (trying again; {} pieces left)", msg, n));
+        tr_logAddWarnTor(tor, fmt::format("metadata error: {}. (trying again; {} pieces left)", msg, n));
```
Comment (mikedld) [inline libtransmission/torrent-magnet.cc:314]: Warning, better message + translate?

File: libtransmission/tr-dht.cc [hunk 4]
```diff
@@ -113,7 +112,7 @@ static void bootstrap_from_name(char const* name, tr_port port, int af)
     addrinfo* info = nullptr;
     if (int const rc = getaddrinfo(name, pp, &hints, &info); rc != 0)
     {
-        tr_logAddNamedWarn(LogName, fmt::format("{}:{}: {}", name, pp, gai_strerror(rc)));
+        tr_logAddWarn(fmt::format("{}:{}: {}", name, pp, gai_strerror(rc)));
```
Comment (mikedld) [inline libtransmission/tr-dht.cc:115]: Warning, better message + translate?

File: libtransmission/peer-io.cc [hunk 5]
```diff
@@ -429,7 +429,7 @@ static void utp_on_read(void* vio, unsigned char const* buf, size_t buflen)
 
     if (rc < 0)
     {
-        tr_logAddNamedError("UTP", "On read evbuffer_add");
+        tr_logAddError("On read evbuffer_add");
```
Comment (mikedld) [inline libtransmission/peer-io.cc:432]: Error, better message + translate?

**Pull request merged**